### PR TITLE
fix: custom mode double initialized xrpl-client

### LINKED ISSUE #645: custom mode has trouble connecting when no port is specified.
The App component re-renders twice and causes the `xrpl-client` to be double instantiated.

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Dynamically determining the VHS network id was causing the sockets to be reinitialized when the response was received. This moves the logic into their own components with their own scope that doesn't trigger everything to be called again.  This would only happen in custom mode which is when network id is unknown upon load.

Fixes #645

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

File: src/containers/App/test/App.test.js [hunk 1]
```diff
@@ -192,4 +199,14 @@ describe('App container', () => {
...
     })
   })
+
+  it('renders custom mode', async () => {
+    process.env.VITE_ENVIRONMENT = 'custom'
+    const wrapper = createWrapper('/s2.ripple.com/')
+    await flushPromises()
+    wrapper.update()
+    // Make sure the sockets aren't double initialized. Called twice. Once for main host and p2p
+    expect(XrplClient).toHaveBeenCalledTimes(2)
```
Comment (JST5000) [inline src/containers/App/test/App.test.js:209]: Can this ever just be 1 if there is no P2P host specified?

**Review by JST5000 — APPROVED** [review]
LGTM - one question, but it's about a test

File: src/containers/App/App.tsx [hunk 2]
```diff
@@ -46,28 +40,9 @@ const App = ({ actions }: AppProps) => {
     return function cleanup() {
       window.removeEventListener('resize', actions.updateViewportDimensions)
       window.removeEventListener('scroll', actions.onScroll)
-
-      socket.close()
-      if (socket.p2pSocket !== undefined) {
-        socket.p2pSocket.close()
-      }
```
Comment (mvadari) [inline src/containers/App/App.tsx:53]: Why was this part removed?

Comment (ckniffen) [reply src/containers/App/App.tsx:53]: It was moved inside the new `<SocketProvider>` component which handles all setup and teardown of the SocketContext.

**Review by mvadari — APPROVED** [review]
